lib/cocoapods/generator/prepare_artifacts_script.rb

```diff
@@ -117,7 +117,7 @@ def script
             fi
 
             local record_artifact=${2:-true}
-            local destination="${TARGET_BUILD_DIR}"
+            local destination="${CONFIGURATION_BUILD_DIR}"
```

Comment:
shouldnt we use something like `BUILT_PRODUCTS_DIR`?

Reply:
Hmm not sure, they seem to be the same in my tests. Here's the docs for each

> Per-configuration Build Products Path (CONFIGURATION_BUILD_DIR)
The base path where build products will be placed during a build for a given configuration. By default, this is set to $(BUILD_DIR)/$(CONFIGURATION).

> BUILT_PRODUCTS_DIR
Identifies the directory under which all the product’s files can be found. This directory contains either product files or symbolic links to them. Run Script build phases can use the value of this build setting as a convenient way to refer to the product files built by one or more targets even when these files are scattered throughout a directory hierarchy (for example, when Deployment Location (DEPLOYMENT_LOCATION) is set to YES.

We've used `${CONFIGURATION_BUILD_DIR}` elsewhere such as https://github.com/CocoaPods/CocoaPods/blob/42741d897eb8ab34f9a1524988f59ff3f7760cc4/lib/cocoapods/generator/embed_frameworks_script.rb#L63

Reply:
sg!

Reply:
I think its good because I think `CONFIGURATION_BUILD_DIR` is on a per-target basis while `BUILT_PRODUCTS_DIR` is global.

Reply:
oh nice, TIL

Reply:
well this is how I read it from what you pasted above :D

Reply:
I initially read "the product's file" as a single product but I realize now it mentions other targets